### ISSUE TITLE
Some unrated recipes don't appear in search for unrated recipes

### ISSUE THREAD

**Issue by tomtjes** [TandoorRecipes/recipes#1939]
### Tandoor Version

1.3.1

### Setup

Docker / Docker-Compose

### Reverse Proxy

No reverse proxy

### Bug description

Only recipes with no cook log or cook logs with `rating: "0"` appear in a search for unrated recipes. However through the API it's possible to create cook log entries with `rating: none`. Those recipes should also be recognized as unrated.

**Comment by vabene1111** [comment]
sorry but if you create invalid data by passing something trough the api thats not supported by the UI than its not bug if some other feature does not behave correctly. 

best case would be if the api did not allow this but sadly the API is not perfect and there are some places where it might allow things that the user interface would not consider valid. I will make this filed not nullable but wont change the search filter.

**Comment by tomtjes** [comment]
Yeah, agree! I just followed the API doc from which it's not clear that the default/unrated state is '0'.

**Comment by tomtjes** [comment]
There seems to be some additional inconsistency here. Recipes with an existing cook log, but a rating of `0` appear in a search for unrated recipes. At the same time, a rating of `0` affects the average rating, i.e. if a recipe has one cook log with a rating of `5` and another cook log with a rating of `0`, the overall rating drops to 2.5. 

The expected behavior would be that you can indeed not rate a recipe when creating a cook log, so I think including recipes with a `0`-rating in unrated recipes is correct. So a `0`-rating should not affect the overall rating.
